addons/account/models/account_payment.py

```diff
@@ -940,13 +940,17 @@ def _compute_display_name(self):
             payment.display_name = payment.move_id.name if payment.move_id.name != '/' else _('Draft Payment')
 
     def copy(self, default=None):
-        if not self.is_internal_transfer:
-            default = {
-                'journal_id': self.journal_id.id,
-                'payment_method_line_id': self.payment_method_line_id.id,
-                **(default or {}),
-            }
-        return super().copy(default=default)
+        copied_payments = self.env['account.payment']
+        for payment_record in self:
+            payment_default = default
+            if not payment_record.is_internal_transfer:
+                payment_default = {
+                    'journal_id': payment_record.journal_id.id,
+                    'payment_method_line_id': payment_record.payment_method_line_id.id,
+                    **(default or {}),
+                }
+            copied_payments += super(AccountPayment, payment_record).copy(default=payment_default)
+        return copied_payments
```

Comment:
Kinda nitpicking but why call it `payment_record` ? self is always records. Just `payment` would be better. Or just `record` altough less explicit imo. Up to you

Also on line 945, you assign `payment_default` to `default` but then you override `payment_default` on line 947 without taking into account its value and adding the values of `default` a second time (to an empty variable).

I believe this should do the same with better performance and readability, please tell me if i'm wrong.
```suggestion
        copied_payments = self.env['account.payment']
        for payment in self:
            payment_default = dict(default or {})
            if not payment.is_internal_transfer:
                payment_default |= {
                    'journal_id': payment.journal_id.id,
                    'payment_method_line_id': payment.payment_method_line_id.id,
                }
            copied_payments += super(AccountPayment, payment).copy(default=payment_default)
        return copied_payments
```

Reply:
Hello @Megaaaaaa, 
Thanks for the feedback.

For the naming i think it might make sense to just use `payment`, i don't mind changing it.

For the second point, correct me if i'm wrong. using `dict(default or {})` will create new `dict` so talking about performance, we in this case will always create new instance but in the current behavior only when `is_internal_transfer` is `True`

Another point, if the record has `journal_id` and/or `payment_method_line_id` we should keep them, in the solution you provided the ones of `payment_record` will override the default ones.

wdyt ?

Reply:
Okay that's fine by me, i didn't take that into account. Nice one :+1: